examples/potions.rs

```diff
@@ -0,0 +1,428 @@
+    }
+
+    for z in -25..25 {
+        for x in -25..25 {
+            layer
+                .chunk
+                .set_block([x, SPAWN_Y, z], BlockState::GRASS_BLOCK);
+        }
+    }
+
+    commands.spawn(layer);
+}
+
+#[allow(clippy::type_complexity)]
+fn init_clients(
+    mut clients: Query<
+        (
+            &mut Client,
+            &mut EntityLayerId,
+            &mut VisibleChunkLayer,
+            &mut VisibleEntityLayers,
+            &mut Position,
+            &mut GameMode,
+        ),
+        Added<Client>,
+    >,
+    layers: Query<Entity, (With<ChunkLayer>, With<EntityLayer>)>,
+) {
+    for (
+        mut client,
+        mut layer_id,
+        mut visible_chunk_layer,
+        mut visible_entity_layers,
+        mut pos,
+        mut game_mode,
+    ) in &mut clients
+    {
+        let layer = layers.single();
+
+        layer_id.0 = layer;
+        visible_chunk_layer.0 = layer;
+        visible_entity_layers.0.insert(layer);
+        pos.set([0.0, SPAWN_Y as f64 + 1.0, 0.0]);
+        *game_mode = GameMode::Survival;
+
+        client.send_chat_message("Welcome to the potions example.".bold());
+        client.send_chat_message("Sneak to apply a random potion effect.".into_text());
+        client.send_chat_message("Note: Some potion effects are not implemented yet.".into_text());
+    }
+}
+
+pub fn add_potion_effect(
+    mut clients: Query<&mut ActiveStatusEffects>,
+    mut events: EventReader<SneakEvent>,
+) {
+    let mut rng = rand::thread_rng();
+    for event in events.read() {
+        if event.state == SneakState::Start {
+            if let Ok(mut status) = clients.get_mut(event.client) {
+                status.apply(
+                    ActiveStatusEffect::from_effect(*StatusEffect::ALL.choose(&mut rng).unwrap())
+                        .with_duration(rng.gen_range(10..1000))
+                        .with_amplifier(rng.gen_range(0..5)),
+                );
+            }
+        }
+    }
+}
+
+const fn parse_uuid_const(uuid: &[u8]) -> Uuid {
+    // unwrap_or can't be used here because it's not a const fn
+    match Uuid::try_parse_ascii(uuid) {
+        Ok(uuid) => uuid,
+        Err(_) => Uuid::nil(),
+    }
+}
+
+const SPEED_UUID: Uuid = parse_uuid_const(b"91AEAA56-376B-4498-935B-2F7F68070635");
+const SLOW_UUID: Uuid = parse_uuid_const(b"7107DE5E-7CE8-4030-940E-514C1F160890");
+const HASTE_UUID: Uuid = parse_uuid_const(b"AF8B6E3F-3328-4C0A-AA36-5BA2BB9DBEF3");
+const MINING_FATIGUE_UUID: Uuid = parse_uuid_const(b"55FCED67-E92A-486E-9800-B47F202C4386");
+const STRENGTH_UUID: Uuid = parse_uuid_const(b"648D7064-6A60-4F59-8ABE-C2C23A6DD7A9");
+const WEAKNESS_UUID: Uuid = parse_uuid_const(b"22653B89-116E-49DC-9B6B-9971489B5BE5");
+const HEALTH_BOOST_UUID: Uuid = parse_uuid_const(b"5D6F0BA2-1186-46AC-B896-C61C5CEE99CC");
+const LUCK_UUID: Uuid = parse_uuid_const(b"03C3C89D-7037-4B42-869F-B146BCB64D2E");
+const UNLUCK_UUID: Uuid = parse_uuid_const(b"CC5AF142-2BD2-4215-B636-2605AED11727");
```

Comment:
Is there a way to extract these constants from Minecraft and generate them in source code?

Reply:
Now that I think about it, I could actually (+ a bit more that might make it easier to implement). I don't have the time to do that now, but I'll do that when I can

Reply:
I'll make another PR for that I think (lmk if you think I shouldn't)